pandas/core/nanops.py

```diff
@@ -554,7 +554,12 @@ def nanprod(values, axis=None, skipna=True):
         values = values.copy()
         values[mask] = 1
     result = values.prod(axis)
-    return _maybe_null_out(result, axis, mask)
+
+    # np compat with >= 1.10
+    allow_all_null = True
+    if not _np_version_under1p10:
```

Comment:
hm it is
I will fix that as it makes more sense to be consistent

it's the tests that can be conditional